src/main/java/rx/Observable.java

```diff
@@ -6705,6 +6706,26 @@ public final Boolean call(T t) {
     }
 
     /**
+     * Allow the an external signal control the amount of data being set through this Observable chain.
+     * When the control Observable emits false (closes the valve) requests upstream are stopped and any
+     * requests from downstream for more data are buffered until the control Observable emits a true
```

Comment:
I added the granularity parameter because the valve would have anything to turn off if the subscriber just requested max long once.